File: q701.cpp
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        if(root==NULL){
            TreeNode*t=new TreeNode(val);
            return t;
        }
        TreeNode*iroot=root;
        while(root!=NULL){
            if(val<root->val){
                if(root->left){
                    root=root->left;
                }
                else{
                    TreeNode*temp=new TreeNode(val);
                    root->left=temp;
                    break;
                }
            }
            else{
                 if(root->right){
                    root=root->right;
                }
                else{
                    TreeNode*temp=new TreeNode(val);
                    root->right=temp;
                    break;
                }
            }
        }
        return iroot;
    }
};

File: q701.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:

        def solve(root, val):
            if root is None:
                return TreeNode(val)
            if root.val > val:
                root.left = solve(root.left, val)
            else:
                root.right = solve(root.right, val)
            return root

        return solve(root, val)
